torch/nn/parallel/_functions.py

```diff
@@ -56,8 +56,12 @@ def forward(ctx, target_device, dim, *inputs):
         assert all(i.device.type != 'cpu' for i in inputs), (
             'Gather function not implemented for CPU tensors'
         )
-        target_device = _get_device_index(target_device, True)
-        ctx.target_device = target_device
+        if (target_device == 'cpu'):
+            target_device = _get_device_index(target_device, True, allow_cpu=True)
+            ctx.target_device = 'cpu'
```

Comment:
Shouldn't this be `ctx.target_device = target_device`? And in general can't the change be reverted to what it already was? 

If not, then is there a need to modify `target_device` for the CPU path?

Reply:
Setting ```ctx.target_device = target_device``` sets ```ctx.target_device ``` to ```-1```, which triggers the Deprecation warning in ```comm.gather```. I'm not sure how to set ```target_device = 'cpu'``` in the cpu case , so I explicitly set the ```ctx.target_device``` to cpu.

Reply:
Removed line 60 because it's unnecessary